test/core/ext/filters/event_engine_client_channel_resolver/resolver_fuzzer.cc

```diff
@@ -142,57 +147,94 @@ class FuzzingResolverEventEngine
-              absl::StatusOr<std::vector<SRVRecord>> response) mutable {
-            runner->Run(
-                [cb = std::move(cb), response = std::move(response)]() mutable {
-                  cb(response);
-                });
-            return EventEngine::DNSResolver::LookupTaskHandle::kInvalid;
-          };
-      return finish(engine_->srv_responses_);
+      int handle = NextHandle();
+      CheckAndSetOrphan(ShouldOrphan::DURING_LOOKUP_SRV);
+      if (!engine_->has_been_orphaned_) {
+        engine_->runner_.Run(
+            [this, cb = std::move(on_resolve), handle]() mutable {
+              if (!HandleExists(handle)) return;
+              DeleteHandle(handle);
+              cb(engine_->srv_responses_);
+              CheckAndSetOrphan(ShouldOrphan::AFTER_LOOKUP_SRV_CALLBACK);
+            });
+      }
+      return {handle, 0};
     }
     LookupTaskHandle LookupTXT(LookupTXTCallback on_resolve,
                                absl::string_view /* name */,
                                Duration /* timeout */) override {
-      auto finish =
-          [cb = std::move(on_resolve), runner = &engine_->runner_](
-              absl::StatusOr<std::vector<std::string>> response) mutable {
-            runner->Run(
-                [cb = std::move(cb), response = std::move(response)]() mutable {
-                  cb(response);
-                });
-            return EventEngine::DNSResolver::LookupTaskHandle::kInvalid;
-          };
-      return finish(engine_->txt_responses_);
+      int handle = NextHandle();
+      CheckAndSetOrphan(ShouldOrphan::DURING_LOOKUP_TXT);
+      if (!engine_->has_been_orphaned_) {
+        engine_->runner_.Run(
+            [this, cb = std::move(on_resolve), handle]() mutable {
+              if (!HandleExists(handle)) return;
+              DeleteHandle(handle);
+              cb(engine_->txt_responses_);
+              CheckAndSetOrphan(ShouldOrphan::AFTER_LOOKUP_TXT_CALLBACK);
+            });
+      }
+      return {handle, 0};
+    }
+    bool CancelLookup(LookupTaskHandle handle) override {
+      int bit_handle = handle.keys[0];
+      if (!HandleExists(bit_handle)) return false;
+      DeleteHandle(bit_handle);
+      return true;
     }
-    bool CancelLookup(LookupTaskHandle /* handle */) override { return false; }
 
    private:
+    int NextHandle() {
+      static int next_handle = 1;
+      if (next_handle == 0) ++next_handle;
+      known_handles |= next_handle;
+      auto ret = next_handle;
+      next_handle <<= 1;
+      return ret;
+    }
+
+    bool HandleExists(int handle) { return known_handles & handle; }
+
+    void DeleteHandle(int handle) { known_handles -= handle; }
+
+    void CheckAndSetOrphan(ShouldOrphan current_pc) {
+      if (engine_->should_orphan_ == current_pc) {
+        *engine_->done_resolving_ = true;
+        engine_->has_been_orphaned_ = true;
+      }
+    }
+
     FuzzingResolverEventEngine* engine_;
+    int known_handles = 0;
```

Comment:
Underscore and documentation added. But I changed it to a hash set, because I realized that even with a uint64_t bitmask (which it wasn't), if the fuzzer generated more than 61 SRV records to re-resolve, task handles would have been reused and odd things would have happened if Orphaning was requested.